pkg/aws/ec2/ec2.go

```diff
@@ -43,14 +44,42 @@ type MetricsAPI interface {
 }
 
 // NewClient returns a new EC2 client
-func NewClient(ec2Client *ec2.Client, metrics MetricsAPI, rateLimit float64, burst int) *Client {
+func NewClient(ec2Client *ec2.Client, metrics MetricsAPI, rateLimit float64, burst int, subnetsFilters []ec2.Filter) *Client {
 	return &Client{
-		ec2Client:  ec2Client,
-		metricsAPI: metrics,
-		limiter:    helpers.NewApiLimiter(metrics, rateLimit, burst),
+		ec2Client:      ec2Client,
+		metricsAPI:     metrics,
+		limiter:        helpers.NewApiLimiter(metrics, rateLimit, burst),
+		subnetsFilters: subnetsFilters,
 	}
 }
 
+// NewSubnetsFilters transforms a map of tags and values and a slice of subnets
+// into a slice of ec2.Filter adequate to filter AWS subnets.
+func NewSubnetsFilters(tags map[string]string, ids []string) []ec2.Filter {
+	var filters []ec2.Filter
+	if len(ids) == 0 {
+		filters = make([]ec2.Filter, 0, len(tags))
+	} else {
+		filters = make([]ec2.Filter, 0, len(tags)+1)
+	}
```

Comment:
We can just take `filters = make([]ec2.Filter, 0, len(tags)+1)`. There's no need to have a branch to to optimize the preallocation by one element.